sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -191,6 +207,27 @@ private ShouldRetryResult shouldRetryOnSessionNotAvailable() {
         }
     }
 
+    private Mono<ShouldRetryResult> shouldRetryOnStaleContainer() {
+        this.staleContainerRetryCount++;
+        if (this.rxCollectionCache == null || this.staleContainerRetryCount > 1) {
```

Comment:
if we pass `rxCollectionCache` in the constructor then you won't need null check here.

Reply:
Please see this comment https://github.com/Azure/azure-sdk-for-java/pull/25811#discussion_r764327955

Reply:
Discussed offline , we are good with this